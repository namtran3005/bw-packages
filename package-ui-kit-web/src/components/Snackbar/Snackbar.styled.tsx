import * as React from 'react';
import styled from 'styled-components';
import { Snackbar as MuiSnackbar } from '@material-ui/core';
import { SnackbarProps as MuiSnackbarProps } from '@material-ui/core/Snackbar';
import { Colors, ColorsType } from '@bitwala-cryptobank-squad/package-theme';

interface RootProps extends MuiSnackbarProps {
  background?: ColorsType;
  foreground?: ColorsType;
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

const RootComponent: React.FunctionComponent<RootProps> = ({
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  background,
  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
  foreground,
  ...cleanProps
}) => <MuiSnackbar {...cleanProps} />;

const defaultBackground = 'primaryBlack';

const Root = styled(RootComponent)`
  &&& {
    & > * {
      border-radius: 0;
      background: ${(props) =>
        Colors.get(props.background || defaultBackground)};
      padding: 0 12px !important;
    }
  }
`;

Root.displayName = 'Snackbar';
RootComponent.displayName = 'SnackbarComponent';

export { Root, RootProps, RootComponent };
