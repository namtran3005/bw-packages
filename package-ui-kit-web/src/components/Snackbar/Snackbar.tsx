import * as React from 'react';
import { SnackbarOrigin } from '@material-ui/core/Snackbar';

import { Typography } from '../Typography';
import { Root, RootProps as SnackbarProps } from './Snackbar.styled';

const Snackbar: React.FunctionComponent<SnackbarProps> = ({
  message,
  anchorOrigin = {
    vertical: 'top',
    horizontal: 'center',
  } as SnackbarOrigin,
  foreground = 'white',
  ...cleanProps
}) => (
  <Root
    message={
      <Typography.Body foreground={foreground}>{message}</Typography.Body>
    }
    {...{
      anchorOrigin,
      ...cleanProps,
    }}
  />
);

export { Snackbar, SnackbarProps };
