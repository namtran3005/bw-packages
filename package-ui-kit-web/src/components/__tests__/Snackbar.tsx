import * as React from 'react';
import { shallow } from 'enzyme';
import { Snackbar } from '../Snackbar';

const defaultProps = {
  open: true,
};

describe('Snackbar', () => {
  it('should render Snackbar', () => {
    const wrapper = shallow(<Snackbar {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
