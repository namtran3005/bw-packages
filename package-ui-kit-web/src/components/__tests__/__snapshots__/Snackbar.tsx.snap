// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar should render Snackbar 1`] = `
<Snackbar
  anchorOrigin={
    Object {
      "horizontal": "center",
      "vertical": "top",
    }
  }
  message={
    <Body
      foreground="white"
    />
  }
  open={true}
/>
`;
